### PR TITLE
Frontend: creation of the profile page

### PR DESCRIPTION
**Pull request by eprei**
# Frontend: creation of the profile page

- start of page profile
- css simplification
- button scaling when is hovered
- CSS overflow problem solved

### CODE REVIEW

File: frontend/src/components/profile/MainProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                        </div>
+                    </div>
+                    <div className={styles.bodyRightSide}>
+                        <FriendList />
+                    </div>
+                </div>
+            </div>
+        </div>
+    )
+}
```
Comment (RobinBurri) [inline frontend/src/components/profile/MainProfile.tsx:37]:
```suggestion
     return (
        <div className={styles.container}>
            <h1>Profile</h1>
            <div className={styles.body}>
                <div className={styles.bodyLeftSide}>
                    <UserInformation
                        picture={PicturePlaceHolder}
                        name="Name placeholder"
                        level={42}
                        TFA={true}
                    />
                    <Statistics />
                    <SeeMatchHistoryBtn />
                </div>
                <div className={styles.bodyRightSide}>
                    <FriendList />
                </div>
            </div>
        </div>
    )
}
```

Comment (RobinBurri) [reply frontend/src/components/profile/MainProfile.tsx:37]: Delete useless div + change Profile title to h1 for better semantic.

File: frontend/src/components/profile/MainProfile.module.css [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    width: 90vw;
+    height: 80vh;
+    min-height: 500px;
+
+    background: var(--color-black-transparent);
+    box-shadow: 0px -4px 16px var(--color-black);
+    border-radius: 28px;
+}
+
+.title {
```
Comment (RobinBurri) [inline frontend/src/components/profile/MainProfile.module.css:17]:
```suggestion
.container h1 {
```

Comment (RobinBurri) [reply frontend/src/components/profile/MainProfile.module.css:17]: to keep Profile title with the same style.

File: frontend/src/components/profile/UserInformation.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                        {name}
+                        <ClickableIcon
+                            icon={IconEditProfile}
+                            onClick={editProfile}
+                        />
+                    </li>
+                    <li>Level {level}</li>
+                    <li>
+                        Two-factor authentication is{' '}
+                        {TFA ? 'activated ' : 'deactivated'}
```
Comment (RobinBurri) [inline frontend/src/components/profile/UserInformation.tsx:47]:
```suggestion
                        Two-factor authentication is
                        {TFA ? ' activated ' : ' deactivated'}
```

Comment (RobinBurri) [reply frontend/src/components/profile/UserInformation.tsx:47]: Move  the space in the ternary

File: frontend/src/components/profile/FriendList.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        picture: PicturePlaceHolder,
+        status: 'online',
+        isFriend: false,
+    },
+]
+
+const FriendList = () => {
+    return (
+        <div className={styles.container}>
+            <div className={styles.subtitle}>Friend list</div>
```
Comment (RobinBurri) [inline frontend/src/components/profile/FriendList.tsx:57]:
```suggestion
            <h3>Friend list</h3>
```

Comment (RobinBurri) [reply frontend/src/components/profile/FriendList.tsx:57]: Let's use something different than a div.

File: frontend/src/components/profile/FriendList.module.css [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+.subtitle {
+    display: flex;
+    justify-content: center;
+
+    font-family: 'Inter';
+    font-style: normal;
+    font-weight: 800;
+    font-size: 16px;
+    line-height: 18px;
+}
```
Comment (RobinBurri) [inline frontend/src/components/profile/FriendList.module.css:15]:
```suggestion
.container h3 {
    font-size: 16px;
}
```

Comment (RobinBurri) [reply frontend/src/components/profile/FriendList.module.css:15]: same effect.

File: frontend/src/components/profile/FriendList.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                            id={friendsRecoveredFromBackend.id}
+                            name={friendsRecoveredFromBackend.name}
+                            picture={friendsRecoveredFromBackend.picture}
+                            status={friendsRecoveredFromBackend.status}
+                            isFriend={friendsRecoveredFromBackend.isFriend}
+                        />
+                    )
+                )}
+            </div>
+            <div className={styles.subtitle}>Other users</div>
```
Comment (RobinBurri) [inline frontend/src/components/profile/FriendList.tsx:72]:
```suggestion
            <h3>Other users</h3>
```

Comment (RobinBurri) [reply frontend/src/components/profile/FriendList.tsx:72]: same here

File: frontend/src/components/profile/Friend.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+            ></span>
+            <span className={styles.nameAndStatus}>
+                <span className={styles.name}>{name}</span>
+                {isFriend ? (
+                    <span className={`${styles.status} ${statusColorClass}`}>
+                        {status}
+                    </span>
+                ) : null}
+            </span>
+        </div>
```
Comment (RobinBurri) [inline frontend/src/components/profile/Friend.tsx:93]:
```suggestion
   <div className={styles.container}>
            <div>
                {
                    <ClickableIcon
                        icon={isFriend ? IconRemoveFriend : IconAddFriend}
                        onClick={isFriend ? removeFriend : addFriend}
                    />
                }
            </div>
            <div
                className={styles.profilePicture}
                style={profilePictureStyle}
            ></div>
            <div className={styles.nameAndStatus}>
                <h3>{name}</h3>
                <p className={`${styles.status} ${statusColorClass}`}>
                    {status}
                </p>
            </div>
        </div>
```

Comment (RobinBurri) [reply frontend/src/components/profile/Friend.tsx:93]: change some divs for more appropriate html tags, like p or h3.
change isFriend ternary to change only the icon and the function and not the element itself.

File: frontend/src/components/profile/Friend.module.css [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    align-items: center;
+}
+
+.number {
+    font-family: 'Inter';
+    font-style: normal;
+    font-weight: 800;
+    font-size: 24px;
+    line-height: 24px;
+}
```
Comment (RobinBurri) [inline frontend/src/components/profile/Friend.module.css:12]:
```suggestion
```

Comment (RobinBurri) [reply frontend/src/components/profile/Friend.module.css:12]: unused

File: frontend/src/components/profile/Friend.module.css [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+    color: var(--color-black-grey);
+}
+
+.nameAndStatus {
+    display: flex;
+    flex-direction: column;
+    gap: 0.2rem;
+}
+
```
Comment (RobinBurri) [inline frontend/src/components/profile/Friend.module.css:57]:
```suggestion
.status {
    font-size: 10px;
    color: var(--color-black-grey);
}

.nameAndStatus {
    display: flex;
    flex-direction: column;
    gap: 0.2rem;
}

.nameAndStatus h3 {
    font-weight: 800;
    font-size: 15px;
    color: #ffffff;
    white-space: nowrap;
}
```

Comment (RobinBurri) [reply frontend/src/components/profile/Friend.module.css:57]: To keep the same style with the changes made in Friend.tsx

File: frontend/src/components/profile/Friend.tsx [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+export interface FriendProps {
+    id: number
+    name: string
+    picture: string
+    status: 'online' | 'offline' | 'playing'
+    isFriend: boolean
+}
+
+const Friend: React.FC<FriendProps> = ({ name, picture, status, isFriend }) => {
```
Comment (RobinBurri) [inline frontend/src/components/profile/Friend.tsx:16]:
```suggestion
const Friend = ({ name, picture, status, isFriend }: FriendProps) => {
```

Comment (RobinBurri) [reply frontend/src/components/profile/Friend.tsx:16]: delete React.FC as it is not recommanded

File: frontend/src/pages/Profile.tsx [hunk 11]
```diff
@@ -1,8 +1,13 @@
...
 const Profile = () => {
     return (
-        <>
-            <h1>Profile</h1>
-        </>
+        <div className={styles.container}>
+            <div className={styles.profile}>
+                <MainProfile />
+            </div>
+        </div>
```
[the remark this answers is not in the document]
Comment (eprei) [reply frontend/src/pages/Profile.tsx:10]: It is a legacy code that is completely disposable as you have rightly pointed out. Thank you!

File: frontend/src/pages/Profile.module.css [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+    flex-direction: column;
+    height: 100%;
+}
+
+.profile {
+    flex: 1;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+}
```
[the remark this answers is not in the document]
Comment (eprei) [reply frontend/src/pages/Profile.module.css:12]: Yes!

File: frontend/src/components/profile/UserInformation.tsx (hunk 3 above)
Comment (eprei) [reply frontend/src/components/profile/UserInformation.tsx:47]: Stylish, thx!

File: frontend/src/components/profile/Friend.tsx (hunk 10 above)
Comment (eprei) [reply frontend/src/components/profile/Friend.tsx:16]: Of course

File: frontend/src/components/profile/Friend.tsx (hunk 7 above)
Comment (eprei) [reply frontend/src/components/profile/Friend.tsx:93]: This is a stylish way to simplify the code and make it more readable. Thanks for showing it to me!

File: frontend/src/components/profile/FriendList.tsx (hunk 4 above)
Comment (eprei) [reply frontend/src/components/profile/FriendList.tsx:57]: Yes!

File: frontend/src/components/profile/FriendList.tsx (hunk 6 above)
Comment (eprei) [reply frontend/src/components/profile/FriendList.tsx:72]: Tout à fait

File: frontend/src/components/profile/ClickableIcon.tsx [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import styles from './ClickableIcon.module.css'
+
+interface ClickableIconProps {
+    icon: string
+    onClick?: () => void
+}
+
+const ClickableIcon: React.FC<ClickableIconProps> = ({ icon, onClick }) => {
```
Comment (RobinBurri) [inline frontend/src/components/profile/ClickableIcon.tsx:9]:
```suggestion
const ClickableIcon = ({ icon, onClick }: ClickableIconProps) => {
```

Comment (RobinBurri) [reply frontend/src/components/profile/ClickableIcon.tsx:9]: No React.FC

File: frontend/src/components/profile/StatisticElement.tsx [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+    icon: string
+    text: string
+    number: number
+}
+
+const StatisticElement: React.FC<StatisticElement> = ({
+    icon,
+    text,
+    number,
+}) => {
```
Comment (RobinBurri) [inline frontend/src/components/profile/StatisticElement.tsx:14]:
```suggestion
const StatisticElement = ({ icon, text, number }: StatisticElement) => {
```

Comment (RobinBurri) [reply frontend/src/components/profile/StatisticElement.tsx:14]: No React.FC

File: frontend/src/components/profile/UserInformation.tsx [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+    level: number
+    TFA: boolean
+}
+
+const UserInformation: React.FC<UserInformation> = ({
+    picture,
+    name,
+    level,
+    TFA,
+}) => {
```
Comment (RobinBurri) [inline frontend/src/components/profile/UserInformation.tsx:18]:
```suggestion
const UserInformation = ({ picture, name, level, TFA }: UserInformation) => {
```

Comment (RobinBurri) [reply frontend/src/components/profile/UserInformation.tsx:18]: Same

File: frontend/src/components/profile/UserInformation.module.css [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+.subtitle {
+    margin-top: 1rem;
+
+    font-family: 'Inter';
+    font-style: normal;
+    font-weight: 800;
+    font-size: 16px;
+    line-height: 18px;
+}
+
```
Comment (RobinBurri) [inline frontend/src/components/profile/UserInformation.module.css:27]:
```suggestion
```

Comment (RobinBurri) [reply frontend/src/components/profile/UserInformation.module.css:27]: unused css

**Review by RobinBurri — APPROVED** [review]
👍

File: frontend/src/components/profile/Friend.tsx [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    const getOpacity = () => {
+        switch (status) {
+            case 'online':
+                return '100%'
+            case 'playing':
+                return '100%'
+            default:
+                return '70%'
```
Comment (t-h2o) [inline frontend/src/components/profile/Friend.tsx:38]: It is probably more explicit
> see: [should-switch-statements-always-contain-a-default-clause](https://stackoverflow.com/questions/4649423/should-switch-statements-always-contain-a-default-clause)
```suggestion
            case 'online':
                return '100%'
            case 'playing':
                return '100%'
            case 'offline':
                return '70%'
```

File: frontend/src/components/profile/Friend.tsx [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+        if (!isFriend) return 'var(--color-black-grey)'
+        switch (status) {
+            case 'online':
+                return 'var(--color-purple)'
+            case 'offline':
+                return 'var(--color-black-grey)'
+            case 'playing':
+                return 'var(--color-mid-green)'
+            default:
+                return 'var(--color-white)'
```
Comment (t-h2o) [inline frontend/src/components/profile/Friend.tsx:27]: Normally, we never go in default ?
```suggestion
```

File: frontend/src/components/profile/Friend.tsx [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+    let statusColorClass = ''
+
+    if (status === 'online') {
+        statusColorClass = styles.online
+    } else if (status === 'playing') {
+        statusColorClass = styles.playing
+    }
+
+    const addFriend = () => {
+        // function to be developed
```
Comment (t-h2o) [inline frontend/src/components/profile/Friend.tsx:60]: TODO keyword with what we will develop ?
> maybe see [do-todo-comments-make-sense](https://softwareengineering.stackexchange.com/questions/125320/do-todo-comments-make-sense)
```suggestion
        // TODO addFriend backend side
```

File: frontend/src/components/profile/MainProfile.module.css [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+
+    font-family: 'Inter';
+    font-style: normal;
+    font-weight: 800;
+    font-size: 32px;
+    line-height: 32px;
+    text-align: center;
+    color: var(--color-white);
+
+    /* background: #3b10e6bd TO DELETE */
```
Comment (t-h2o) [inline frontend/src/components/profile/MainProfile.module.css:28]: is it still useful ?

File: frontend/src/components/profile/MainProfile.module.css [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+
+    width: 100%;
+    height: 100%;
+
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+
+    /* background: #159c5650   TO DELETE */
```
Comment (t-h2o) [inline frontend/src/components/profile/MainProfile.module.css:42]: is it still useful ?

File: frontend/src/components/profile/SeeMatchHistoryBtn.tsx [hunk 22]
```diff
@@ -0,0 +1,12 @@
+import { Link } from 'react-router-dom'
+import styles from './SeeMatchHistoryBtn.module.css'
+
+const SeeMatchHistory = () => {
```
Comment (t-h2o) [inline frontend/src/components/profile/SeeMatchHistoryBtn.tsx:4]: Should we name this as the filename ?
```suggestion
const SeeMatchHistoryBtn = () => {
```

**Review by t-h2o — COMMENTED** [review]
:tropical_fish:

**Review by shaolin-peanut — APPROVED** [review]
nice!

File: frontend/src/components/profile/Friend.tsx (hunk 18 above)
Comment (eprei) [reply frontend/src/components/profile/Friend.tsx:27]: You are absolutely right.

File: frontend/src/components/profile/Friend.tsx (hunk 17 above)
Comment (eprei) [reply frontend/src/components/profile/Friend.tsx:36]: Absolutely, good review!

File: frontend/src/components/profile/SeeMatchHistoryBtn.tsx (hunk 22 above)
Comment (eprei) [reply frontend/src/components/profile/SeeMatchHistoryBtn.tsx:4]: Yes !

File: frontend/src/components/profile/MainProfile.module.css (hunk 21 above)
Comment (eprei) [reply frontend/src/components/profile/MainProfile.module.css:42]: It's legacy code from when I was starting out with CSS and didn't know any other way to debug. I'll quickly remove it in the next commit!

File: frontend/src/components/profile/MainProfile.module.css (hunk 20 above)
Comment (eprei) [reply frontend/src/components/profile/MainProfile.module.css:28]: It's legacy code from when I was starting out with CSS and didn't know any other way to debug. I'll quickly remove it in the next commit!

File: frontend/src/components/profile/Friend.module.css [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+.nameAndStatus h3 {
+    font-weight: 800;
+    font-size: 15px;
+    color: white;
```
Comment (t-h2o) [inline frontend/src/components/profile/Friend.module.css:30]: We have the `--color-white` variable
> see #41 
```suggestion
    color: var(--color-white);
```

File: frontend/src/components/profile/Friend.tsx [hunk 24]
```diff
@@ -0,0 +1,91 @@
...
+
+    const getOpacity = () => {
+        switch (status) {
+            case 'online':
+                return '100%'
+            case 'playing':
+                return '100%'
+            case 'offline':
+                return '70%'
+        }
```
Comment (t-h2o) [inline frontend/src/components/profile/Friend.tsx:37]: consistency, should we use the same `case` order ?
1. online
2. playing
3. offline

Comment (t-h2o) [inline frontend/src/components/profile/Friend.tsx:37]: Should we use `break` with the `switch` ?

File: frontend/src/components/profile/Friend.tsx [hunk 25]
```diff
@@ -0,0 +1,91 @@
...
+        statusColorClass = styles.playing
+    }
+
+    const addFriend = () => {
+        // TODO addFriend backend side
+        console.log('Friend added')
+    }
+
+    const removeFriend = () => {
+        // function to be developed
```
Comment (t-h2o) [inline frontend/src/components/profile/Friend.tsx:63]: use `TODO` keyword followed by a little explanation for each ?
```bash
grep -r "function to be developed" frontend/src
```

File: frontend/src/components/profile/SeeMatchHistoryBtn.tsx [hunk 26]
```diff
@@ -0,0 +1,12 @@
...
+
+const SeeMatchHistoryBtn = () => {
+    return (
+        <Link to="/history" className={styles.btn}>
+            see match history
+        </Link>
+    )
+}
+
+export default SeeMatchHistory
```
Comment (t-h2o) [inline frontend/src/components/profile/SeeMatchHistoryBtn.tsx:12]: So add Btn here too ?
```suggestion
export default SeeMatchHistoryBtn
```

**Review by t-h2o — COMMENTED** [review]
:moon:

File: frontend/src/components/profile/Friend.tsx (hunk 24 above)
Comment (RobinBurri) [reply frontend/src/components/profile/Friend.tsx:37]: Not needed if it's in a function:
https://tutorial.eyehunts.com/js/javascript-switch-return-statement-example-code/

File: frontend/src/components/profile/SeeMatchHistoryBtn.tsx (hunk 26 above)
Comment (eprei) [reply frontend/src/components/profile/SeeMatchHistoryBtn.tsx:12]: Of course!

File: frontend/src/components/profile/Friend.tsx (hunk 25 above)
Comment (eprei) [reply frontend/src/components/profile/Friend.tsx]: Done

File: frontend/src/components/profile/Friend.tsx (hunk 24 above)
Comment (eprei) [reply frontend/src/components/profile/Friend.tsx]: Done

File: frontend/src/components/profile/Friend.module.css (hunk 23 above)
Comment (t-h2o) [reply frontend/src/components/profile/Friend.module.css]: done : 2a51fcc166d15a60233ac96040cfc214649ead15

File: frontend/src/components/profile/Friend.tsx (hunk 25 above)
Comment (t-h2o) [reply frontend/src/components/profile/Friend.tsx]: 6d5165d1682bded69fb8b1f975faeb404896016f

**Review by t-h2o — APPROVED** [review]
:fondue:

**Review by fire-poy — APPROVED** [review]

**Pull request merged**